src/DeeplClient.php

```diff
@@ -24,11 +24,11 @@
 
 class DeeplClient implements DeeplClientInterface
 {
-    private $deeplRequestFactory;
+    private DeeplRequestFactoryInterface $deeplRequestFactory;
```

Comment:
i think we can start with constructor property promotion, and removing php7.4 support

Reply:
K. I'd prefer to update the contructors in a separate PR after this one to keep the currect one small and just stan related.